2019-04-17 12:10:46.842 -07:00 [INF] Using the default authentication scheme Identity.Application for IdentityServer
2019-04-17 12:10:46.938 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for authentication
2019-04-17 12:10:46.939 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-in
2019-04-17 12:10:46.939 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-out
2019-04-17 12:10:46.940 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for challenge
2019-04-17 12:10:46.940 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for forbid
2019-04-17 12:10:47.939 -07:00 [DBG] Starting grant removal
2019-04-17 12:10:50.506 -07:00 [DBG] Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslStream.BeginAuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken, AsyncCallback asyncCallback, Object asyncState)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsServerAsync>b__51_0(SslServerAuthenticationOptions arg1, CancellationToken arg2, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2](Func`5 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state)
   at System.Net.Security.SslStream.AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionAdapter.InnerOnConnectionAsync(ConnectionAdapterContext context)
2019-04-17 12:10:50.828 -07:00 [DBG] Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.PartialFrameCallback(AsyncProtocolRequest asyncRequest)
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Security.SslState.ThrowIfExceptional()
   at System.Net.Security.SslState.InternalEndProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result)
   at System.Net.Security.SslStream.EndAuthenticateAsServer(IAsyncResult asyncResult)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsServerAsync>b__51_1(IAsyncResult iar)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionAdapter.InnerOnConnectionAsync(ConnectionAdapterContext context)
2019-04-17 12:10:55.884 -07:00 [DBG] Login Url: /Account/Login
2019-04-17 12:10:55.889 -07:00 [DBG] Login Return Url Parameter: ReturnUrl
2019-04-17 12:10:55.889 -07:00 [DBG] Logout Url: /Account/Logout
2019-04-17 12:10:55.890 -07:00 [DBG] ConsentUrl Url: /consent
2019-04-17 12:10:55.890 -07:00 [DBG] Consent Return Url Parameter: returnUrl
2019-04-17 12:10:55.891 -07:00 [DBG] Error Url: /home/error
2019-04-17 12:10:55.893 -07:00 [DBG] Error Id Parameter: errorId
2019-04-17 12:15:26.241 -07:00 [DBG] Request path /.well-known/openid-configuration matched to endpoint type Discovery
2019-04-17 12:15:26.324 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryEndpoint
2019-04-17 12:15:26.328 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryEndpoint for /.well-known/openid-configuration
2019-04-17 12:15:26.335 -07:00 [DBG] Start discovery request
2019-04-17 12:15:28.004 -07:00 [ERR] Failed executing DbCommand (251ms) [Parameters=[], CommandType='"Text"', CommandTimeout='30']
SELECT [x].[Id], [x].[Created], [x].[Description], [x].[DisplayName], [x].[Emphasize], [x].[Enabled], [x].[Name], [x].[NonEditable], [x].[Required], [x].[ShowInDiscoveryDocument], [x].[Updated]
FROM [IdentityResources] AS [x]
ORDER BY [x].[Id]
System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
ClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765
Error Number:208,State:1,Class:16
2019-04-17 12:15:28.037 -07:00 [ERR] An exception occurred in the database while iterating the results of a query for context type 'IdentityServer4.EntityFramework.DbContexts.ConfigurationDbContext'.
System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider._TrackEntities[TOut,TIn](IEnumerable`1 results, QueryContext queryContext, IList`1 entityTrackingInfos, IList`1 entityAccessors)+MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
ClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765
Error Number:208,State:1,Class:16
System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider._TrackEntities[TOut,TIn](IEnumerable`1 results, QueryContext queryContext, IList`1 entityTrackingInfos, IList`1 entityAccessors)+MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
ClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765
Error Number:208,State:1,Class:16
2019-04-17 12:15:28.128 -07:00 [INF] {"Details":"System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.\r\n   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)\r\n   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)\r\n   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)\r\n   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)\r\n   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()\r\n   at System.Data.SqlClient.SqlDataReader.get_MetaData()\r\n   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)\r\n   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)\r\n   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)\r\n   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)\r\n   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)\r\n   at System.Data.Common.DbCommand.ExecuteReader()\r\n   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)\r\n   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)\r\n   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)\r\n   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)\r\n   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()\r\n   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()\r\n   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider._TrackEntities[TOut,TIn](IEnumerable`1 results, QueryContext queryContext, IList`1 entityTrackingInfos, IList`1 entityAccessors)+MoveNext()\r\n   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()\r\n   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)\r\n   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)\r\n   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)\r\n   at IdentityServer4.EntityFramework.Stores.ResourceStore.GetAllResourcesAsync() in C:\\local\\identity\\server4\\EntityFramework.Storage\\src\\Stores\\ResourceStore.cs:line 138\r\n   at IdentityServer4.Stores.IResourceStoreExtensions.GetAllEnabledResourcesAsync(IResourceStore store) in C:\\local\\identity\\server4\\IdentityServer4\\src\\Extensions\\IResourceStoreExtensions.cs:line 105\r\n   at IdentityServer4.ResponseHandling.DiscoveryResponseGenerator.CreateDiscoveryDocumentAsync(String baseUrl, String issuerUri) in C:\\local\\identity\\server4\\IdentityServer4\\src\\ResponseHandling\\Default\\DiscoveryResponseGenerator.cs:line 169\r\n   at IdentityServer4.Endpoints.DiscoveryEndpoint.ProcessAsync(HttpContext context) in C:\\local\\identity\\server4\\IdentityServer4\\src\\Endpoints\\DiscoveryEndpoint.cs:line 58\r\n   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events) in C:\\local\\identity\\server4\\IdentityServer4\\src\\Hosting\\IdentityServerMiddleware.cs:line 54\r\nClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765\r\nError Number:208,State:1,Class:16","Category":"Error","Name":"Unhandled Exception","EventType":"Error","Id":3000,"Message":"Invalid object name 'IdentityResources'.","ActivityId":"0HLM3D1J9FN1H:00000002","TimeStamp":"2019-04-17T19:15:28.0000000Z","ProcessId":6816,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"UnhandledExceptionEvent"}
2019-04-17 12:15:28.152 -07:00 [FTL] Unhandled exception: Invalid object name 'IdentityResources'.
System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider._TrackEntities[TOut,TIn](IEnumerable`1 results, QueryContext queryContext, IList`1 entityTrackingInfos, IList`1 entityAccessors)+MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at IdentityServer4.EntityFramework.Stores.ResourceStore.GetAllResourcesAsync() in C:\local\identity\server4\EntityFramework.Storage\src\Stores\ResourceStore.cs:line 138
   at IdentityServer4.Stores.IResourceStoreExtensions.GetAllEnabledResourcesAsync(IResourceStore store) in C:\local\identity\server4\IdentityServer4\src\Extensions\IResourceStoreExtensions.cs:line 105
   at IdentityServer4.ResponseHandling.DiscoveryResponseGenerator.CreateDiscoveryDocumentAsync(String baseUrl, String issuerUri) in C:\local\identity\server4\IdentityServer4\src\ResponseHandling\Default\DiscoveryResponseGenerator.cs:line 169
   at IdentityServer4.Endpoints.DiscoveryEndpoint.ProcessAsync(HttpContext context) in C:\local\identity\server4\IdentityServer4\src\Endpoints\DiscoveryEndpoint.cs:line 58
   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events) in C:\local\identity\server4\IdentityServer4\src\Hosting\IdentityServerMiddleware.cs:line 54
ClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765
Error Number:208,State:1,Class:16
2019-04-17 12:15:28.598 -07:00 [ERR] An unhandled exception has occurred while executing the request.
System.Data.SqlClient.SqlException (0x80131904): Invalid object name 'IdentityResources'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider._TrackEntities[TOut,TIn](IEnumerable`1 results, QueryContext queryContext, IList`1 entityTrackingInfos, IList`1 entityAccessors)+MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at IdentityServer4.EntityFramework.Stores.ResourceStore.GetAllResourcesAsync() in C:\local\identity\server4\EntityFramework.Storage\src\Stores\ResourceStore.cs:line 138
   at IdentityServer4.Stores.IResourceStoreExtensions.GetAllEnabledResourcesAsync(IResourceStore store) in C:\local\identity\server4\IdentityServer4\src\Extensions\IResourceStoreExtensions.cs:line 105
   at IdentityServer4.ResponseHandling.DiscoveryResponseGenerator.CreateDiscoveryDocumentAsync(String baseUrl, String issuerUri) in C:\local\identity\server4\IdentityServer4\src\ResponseHandling\Default\DiscoveryResponseGenerator.cs:line 169
   at IdentityServer4.Endpoints.DiscoveryEndpoint.ProcessAsync(HttpContext context) in C:\local\identity\server4\IdentityServer4\src\Endpoints\DiscoveryEndpoint.cs:line 58
   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events) in C:\local\identity\server4\IdentityServer4\src\Hosting\IdentityServerMiddleware.cs:line 54
   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events) in C:\local\identity\server4\IdentityServer4\src\Hosting\IdentityServerMiddleware.cs:line 69
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.BaseUrlMiddleware.Invoke(HttpContext context) in C:\local\identity\server4\IdentityServer4\src\Hosting\BaseUrlMiddleware.cs:line 36
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
ClientConnectionId:2706d065-2686-4e0c-ad1a-ed767a966765
Error Number:208,State:1,Class:16
2019-04-17 12:16:49.898 -07:00 [INF] Using the default authentication scheme Identity.Application for IdentityServer
2019-04-17 12:16:49.984 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for authentication
2019-04-17 12:16:49.985 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-in
2019-04-17 12:16:49.985 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-out
2019-04-17 12:16:49.985 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for challenge
2019-04-17 12:16:49.985 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for forbid
2019-04-17 12:16:50.381 -07:00 [DBG] Starting grant removal
2019-04-17 12:16:51.368 -07:00 [DBG] Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslStream.BeginAuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken, AsyncCallback asyncCallback, Object asyncState)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsServerAsync>b__51_0(SslServerAuthenticationOptions arg1, CancellationToken arg2, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2](Func`5 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state)
   at System.Net.Security.SslStream.AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionAdapter.InnerOnConnectionAsync(ConnectionAdapterContext context)
2019-04-17 12:16:52.204 -07:00 [DBG] Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslStream.BeginAuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken, AsyncCallback asyncCallback, Object asyncState)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsServerAsync>b__51_0(SslServerAuthenticationOptions arg1, CancellationToken arg2, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2](Func`5 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state)
   at System.Net.Security.SslStream.AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionAdapter.InnerOnConnectionAsync(ConnectionAdapterContext context)
2019-04-17 12:16:54.563 -07:00 [DBG] Login Url: /Account/Login
2019-04-17 12:16:54.566 -07:00 [DBG] Login Return Url Parameter: ReturnUrl
2019-04-17 12:16:54.567 -07:00 [DBG] Logout Url: /Account/Logout
2019-04-17 12:16:54.567 -07:00 [DBG] ConsentUrl Url: /consent
2019-04-17 12:16:54.568 -07:00 [DBG] Consent Return Url Parameter: returnUrl
2019-04-17 12:16:54.575 -07:00 [DBG] Error Url: /home/error
2019-04-17 12:16:54.576 -07:00 [DBG] Error Id Parameter: errorId
2019-04-17 12:19:55.113 -07:00 [DBG] Request path /.well-known/openid-configuration matched to endpoint type Discovery
2019-04-17 12:19:55.208 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryEndpoint
2019-04-17 12:19:55.212 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryEndpoint for /.well-known/openid-configuration
2019-04-17 12:19:55.217 -07:00 [DBG] Start discovery request
2019-04-17 12:19:57.277 -07:00 [DBG] Found ["openid","profile","email","custom.profile","roles","api1","api2.full_access","api2.read_only"] as all scopes in database
2019-04-17 12:20:10.275 -07:00 [DBG] Request path /connect/userinfo matched to endpoint type Userinfo
2019-04-17 12:20:10.280 -07:00 [DBG] Endpoint enabled: Userinfo, successfully created handler: IdentityServer4.Endpoints.UserInfoEndpoint
2019-04-17 12:20:10.281 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.UserInfoEndpoint for /connect/userinfo
2019-04-17 12:20:10.293 -07:00 [DBG] Start userinfo request
2019-04-17 12:20:10.298 -07:00 [DBG] Bearer token not found
2019-04-17 12:20:10.300 -07:00 [ERR] No access token found.
2019-04-17 12:20:21.972 -07:00 [DBG] Request path /.well-known/openid-configuration/jwks matched to endpoint type Discovery
2019-04-17 12:20:21.974 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryKeyEndpoint
2019-04-17 12:20:21.975 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryKeyEndpoint for /.well-known/openid-configuration/jwks
2019-04-17 12:20:21.986 -07:00 [DBG] Start key discovery request
2019-04-17 12:20:29.358 -07:00 [DBG] Request path /connect/revocation matched to endpoint type Revocation
2019-04-17 12:20:29.367 -07:00 [DBG] Endpoint enabled: Revocation, successfully created handler: IdentityServer4.Endpoints.TokenRevocationEndpoint
2019-04-17 12:20:29.368 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.TokenRevocationEndpoint for /connect/revocation
2019-04-17 12:20:29.375 -07:00 [WRN] Invalid HTTP method
2019-04-17 12:20:35.597 -07:00 [DBG] Request path /connect/userinfo matched to endpoint type Userinfo
2019-04-17 12:20:35.599 -07:00 [DBG] Endpoint enabled: Userinfo, successfully created handler: IdentityServer4.Endpoints.UserInfoEndpoint
2019-04-17 12:20:35.655 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.UserInfoEndpoint for /connect/userinfo
2019-04-17 12:20:35.657 -07:00 [DBG] Start userinfo request
2019-04-17 12:20:35.658 -07:00 [DBG] Bearer token not found
2019-04-17 12:20:35.664 -07:00 [ERR] No access token found.
2019-04-17 12:34:02.172 -07:00 [INF] Using the default authentication scheme Identity.Application for IdentityServer
2019-04-17 12:34:02.388 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for authentication
2019-04-17 12:34:02.510 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-in
2019-04-17 12:34:02.520 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-out
2019-04-17 12:34:02.528 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for challenge
2019-04-17 12:34:02.592 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for forbid
2019-04-17 12:34:04.659 -07:00 [DBG] Starting grant removal
2019-04-17 12:34:07.138 -07:00 [DBG] Login Url: /Account/Login
2019-04-17 12:34:07.145 -07:00 [DBG] Login Return Url Parameter: ReturnUrl
2019-04-17 12:34:07.146 -07:00 [DBG] Logout Url: /Account/Logout
2019-04-17 12:34:07.147 -07:00 [DBG] ConsentUrl Url: /consent
2019-04-17 12:34:07.148 -07:00 [DBG] Consent Return Url Parameter: returnUrl
2019-04-17 12:34:07.148 -07:00 [DBG] Error Url: /home/error
2019-04-17 12:34:07.149 -07:00 [DBG] Error Id Parameter: errorId
2019-04-17 12:34:07.187 -07:00 [DBG] Request path /.well-known/openid-configuration matched to endpoint type Discovery
2019-04-17 12:34:07.272 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryEndpoint
2019-04-17 12:34:07.337 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryEndpoint for /.well-known/openid-configuration
2019-04-17 12:34:07.442 -07:00 [DBG] Start discovery request
2019-04-17 12:34:11.734 -07:00 [DBG] Found ["openid","profile","email","custom.profile","roles","api1","api2.full_access","api2.read_only"] as all scopes in database
2019-04-17 12:34:12.808 -07:00 [DBG] Request path /.well-known/openid-configuration/jwks matched to endpoint type Discovery
2019-04-17 12:34:12.811 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryKeyEndpoint
2019-04-17 12:34:12.813 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryKeyEndpoint for /.well-known/openid-configuration/jwks
2019-04-17 12:34:12.817 -07:00 [DBG] Start key discovery request
2019-04-17 12:34:12.938 -07:00 [DBG] Request path /connect/authorize matched to endpoint type Authorize
2019-04-17 12:34:12.942 -07:00 [DBG] Endpoint enabled: Authorize, successfully created handler: IdentityServer4.Endpoints.AuthorizeEndpoint
2019-04-17 12:34:12.943 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.AuthorizeEndpoint for /connect/authorize
2019-04-17 12:34:12.947 -07:00 [DBG] Start authorize request
2019-04-17 12:34:12.960 -07:00 [DBG] No user present in authorize request
2019-04-17 12:34:12.979 -07:00 [DBG] Start authorize request protocol validation
2019-04-17 12:34:13.338 -07:00 [DBG] mvcclientweb found in database: false
2019-04-17 12:34:13.361 -07:00 [ERR] Unknown client or not enabled: mvcclientweb
{"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"mvcclientweb","redirect_uri":"https://localhost:6002/signin-oidc","response_mode":"form_post","response_type":"code id_token","scope":"openid profile","state":"OpenIdConnect.AuthenticationProperties=wD_U9-at_-EjJw4XQbflqsBZn7wB571LMlFN2t4fyVOwvWB8O-wtM49-O1gN7D4zAUkD6OOI3aE1nf7ZnS7zZB--rBpHbQ0GgFKqAvfuSqMLnfJr1NxTycx_PDu3AcFplqrR9KnuF9xlCGIO8LfXSjNHhKUsQ_AVnXiJjEcaJ50eBiluf8XY6btyDC48Fumq2aUEGijbv7TQrf5HhzqK6g","nonce":"636911264528841681.YzJjMzQ3NGEtNjk4Yi00MjY1LTkxNjktYTFkMjNkODU3ODVjMDU4Mjg3OTUtNjI4NC00Nzc2LTliNjItYTA3YTY0NDAxZDQ4","x-client-SKU":"ID_NET451","x-client-ver":"5.2.1.0"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:34:13.376 -07:00 [ERR] Request validation failed
2019-04-17 12:34:13.378 -07:00 [INF] {"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"mvcclientweb","redirect_uri":"https://localhost:6002/signin-oidc","response_mode":"form_post","response_type":"code id_token","scope":"openid profile","state":"OpenIdConnect.AuthenticationProperties=wD_U9-at_-EjJw4XQbflqsBZn7wB571LMlFN2t4fyVOwvWB8O-wtM49-O1gN7D4zAUkD6OOI3aE1nf7ZnS7zZB--rBpHbQ0GgFKqAvfuSqMLnfJr1NxTycx_PDu3AcFplqrR9KnuF9xlCGIO8LfXSjNHhKUsQ_AVnXiJjEcaJ50eBiluf8XY6btyDC48Fumq2aUEGijbv7TQrf5HhzqK6g","nonce":"636911264528841681.YzJjMzQ3NGEtNjk4Yi00MjY1LTkxNjktYTFkMjNkODU3ODVjMDU4Mjg3OTUtNjI4NC00Nzc2LTliNjItYTA3YTY0NDAxZDQ4","x-client-SKU":"ID_NET451","x-client-ver":"5.2.1.0"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:34:13.483 -07:00 [INF] {"ClientId":"mvcclientweb","ClientName":null,"RedirectUri":null,"Endpoint":"Authorize","SubjectId":null,"Scopes":"","GrantType":null,"Error":"unauthorized_client","ErrorDescription":"Unknown client or client not enabled","Category":"Token","Name":"Token Issued Failure","EventType":"Failure","Id":2001,"Message":null,"ActivityId":"0HLM3DEJD2MJ4:00000002","TimeStamp":"2019-04-17T19:34:13.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"TokenIssuedFailureEvent"}
2019-04-17 12:35:00.906 -07:00 [DBG] Request path /.well-known/openid-configuration matched to endpoint type Discovery
2019-04-17 12:35:00.909 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryEndpoint
2019-04-17 12:35:00.910 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryEndpoint for /.well-known/openid-configuration
2019-04-17 12:35:00.912 -07:00 [DBG] Start discovery request
2019-04-17 12:35:01.262 -07:00 [DBG] Found ["openid","profile","email","custom.profile","roles","api1","api2.full_access","api2.read_only"] as all scopes in database
2019-04-17 12:35:01.455 -07:00 [DBG] Request path /.well-known/openid-configuration/jwks matched to endpoint type Discovery
2019-04-17 12:35:01.466 -07:00 [DBG] Endpoint enabled: Discovery, successfully created handler: IdentityServer4.Endpoints.DiscoveryKeyEndpoint
2019-04-17 12:35:01.468 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.DiscoveryKeyEndpoint for /.well-known/openid-configuration/jwks
2019-04-17 12:35:01.469 -07:00 [DBG] Start key discovery request
2019-04-17 12:35:01.777 -07:00 [DBG] CORS request made for path: /connect/authorize from origin: https://localhost:6001 but was ignored because path was not for an allowed IdentityServer CORS endpoint
2019-04-17 12:35:01.781 -07:00 [DBG] Request path /connect/authorize matched to endpoint type Authorize
2019-04-17 12:35:01.784 -07:00 [DBG] Endpoint enabled: Authorize, successfully created handler: IdentityServer4.Endpoints.AuthorizeEndpoint
2019-04-17 12:35:01.785 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.AuthorizeEndpoint for /connect/authorize
2019-04-17 12:35:01.786 -07:00 [DBG] Start authorize request
2019-04-17 12:35:01.809 -07:00 [DBG] No user present in authorize request
2019-04-17 12:35:01.813 -07:00 [DBG] Start authorize request protocol validation
2019-04-17 12:35:02.447 -07:00 [DBG] clientwebapp found in database: true
2019-04-17 12:35:02.491 -07:00 [ERR] Invalid client configuration for client clientwebapp: No redirect URI configured.
2019-04-17 12:35:02.493 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":"Client Web App","Category":"Error","Name":"Invalid Client Configuration","EventType":"Error","Id":3001,"Message":"No redirect URI configured.","ActivityId":"0HLM3DEJD2MJ4:00000004","TimeStamp":"2019-04-17T19:35:02.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"InvalidClientConfigurationEvent"}
2019-04-17 12:35:02.495 -07:00 [ERR] Unknown client or not enabled: clientwebapp
{"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"clientwebapp","redirect_uri":"https://localhost:6001/signin-oidc","response_type":"code id_token","scope":"openid profile resource.api1 offline_access","response_mode":"form_post","nonce":"636911265015407076.MWZhMzY4OTktNmUxZC00ZDJmLTkyNDctZGIyODVlNWRiNmU1NTUxOWE1NjgtZTQ1NS00YTA4LWIzNjItMmQ4MzEyNDgwMTE4","state":"CfDJ8EEgl5mjJYhFtS8DfJca5DNT132yaqmrObFrlD2kuiX7OW12vdoEOmBdBk_yHzhTz_H5HhoXvqiTroJ9pgpuzZaRiKtvwPKQUEbA9D6ZZfzpy36rN7VzbnxNk7p2Pzn_Pe_FWV-fRk4UQvQppv-Qi4vQD_CShJADyfiBIPA7TKRUMBKPzISCAZsIsBcbHvkrTEKnkaR82_4hS6x033xvBYF2D0b9WXEXfSRmxz8m8SavOw9vX7EjmKxt6i1gHeDI7hd-Q3-99mG_KWG_n5YHhr-xbUGVsBOIZdOgQZ-jO-RCXSIjjdc8p1JRDXI0th3FUKC-EkoaBOZ3jpkDtqXr77M"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:35:02.507 -07:00 [ERR] Request validation failed
2019-04-17 12:35:02.507 -07:00 [INF] {"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"clientwebapp","redirect_uri":"https://localhost:6001/signin-oidc","response_type":"code id_token","scope":"openid profile resource.api1 offline_access","response_mode":"form_post","nonce":"636911265015407076.MWZhMzY4OTktNmUxZC00ZDJmLTkyNDctZGIyODVlNWRiNmU1NTUxOWE1NjgtZTQ1NS00YTA4LWIzNjItMmQ4MzEyNDgwMTE4","state":"CfDJ8EEgl5mjJYhFtS8DfJca5DNT132yaqmrObFrlD2kuiX7OW12vdoEOmBdBk_yHzhTz_H5HhoXvqiTroJ9pgpuzZaRiKtvwPKQUEbA9D6ZZfzpy36rN7VzbnxNk7p2Pzn_Pe_FWV-fRk4UQvQppv-Qi4vQD_CShJADyfiBIPA7TKRUMBKPzISCAZsIsBcbHvkrTEKnkaR82_4hS6x033xvBYF2D0b9WXEXfSRmxz8m8SavOw9vX7EjmKxt6i1gHeDI7hd-Q3-99mG_KWG_n5YHhr-xbUGVsBOIZdOgQZ-jO-RCXSIjjdc8p1JRDXI0th3FUKC-EkoaBOZ3jpkDtqXr77M"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:35:02.514 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":null,"RedirectUri":null,"Endpoint":"Authorize","SubjectId":null,"Scopes":"","GrantType":null,"Error":"unauthorized_client","ErrorDescription":"Unknown client or client not enabled","Category":"Token","Name":"Token Issued Failure","EventType":"Failure","Id":2001,"Message":null,"ActivityId":"0HLM3DEJD2MJ4:00000004","TimeStamp":"2019-04-17T19:35:02.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"TokenIssuedFailureEvent"}
2019-04-17 12:35:22.401 -07:00 [DBG] CORS request made for path: /connect/authorize from origin: https://localhost:6001 but was ignored because path was not for an allowed IdentityServer CORS endpoint
2019-04-17 12:35:22.432 -07:00 [DBG] Request path /connect/authorize matched to endpoint type Authorize
2019-04-17 12:35:22.433 -07:00 [DBG] Endpoint enabled: Authorize, successfully created handler: IdentityServer4.Endpoints.AuthorizeEndpoint
2019-04-17 12:35:22.434 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.AuthorizeEndpoint for /connect/authorize
2019-04-17 12:35:22.435 -07:00 [DBG] Start authorize request
2019-04-17 12:35:22.565 -07:00 [DBG] No user present in authorize request
2019-04-17 12:35:22.571 -07:00 [DBG] Start authorize request protocol validation
2019-04-17 12:35:23.112 -07:00 [DBG] clientwebapp found in database: true
2019-04-17 12:35:23.128 -07:00 [ERR] Invalid client configuration for client clientwebapp: No redirect URI configured.
2019-04-17 12:35:23.129 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":"Client Web App","Category":"Error","Name":"Invalid Client Configuration","EventType":"Error","Id":3001,"Message":"No redirect URI configured.","ActivityId":"0HLM3DEJD2MJ4:00000006","TimeStamp":"2019-04-17T19:35:23.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"InvalidClientConfigurationEvent"}
2019-04-17 12:35:23.131 -07:00 [ERR] Unknown client or not enabled: clientwebapp
{"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"clientwebapp","redirect_uri":"https://localhost:6001/signin-oidc","response_type":"code id_token","scope":"openid profile resource.api1 offline_access","response_mode":"form_post","nonce":"636911265221205762.NjRkNzczZGUtNzZiYS00MDAzLThiZDctMGEzYzAzMTczZDIxM2ZiZWQ1ZTAtNDM3MC00MjY5LWI3NTAtMjFlZGZiOWM2NmJj","state":"CfDJ8EEgl5mjJYhFtS8DfJca5DMMwYWMNQBxhUpdgEv3-jVDQ0lRjYpE1XnjX6Qfw_biWcbI4wp4xUi4n67096kmnRwzqRtgwgktn5Y5OYCg9RY22_lEAju5XG6uULAMBhSSE3dlvT-ZObxL8lVpFk1wi-WLUXbRAO5b9gEkjkiJObbgpzBRlY5EAPSfTjMOMzhYvITiQ-xrW9EYMual5-eRG9Br8fxqu4RIzK0HeiMUnTAboihUkB_eEa9p9oDQ-Auo9eCgMUsU7LDd5UKM8GBgIxfpKzHnyMND16q8ZgeV1fCvLCyNcj19wrsCLSgKzjHZ2u1anabVmX9BjoDQWVsV7kE"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:35:23.135 -07:00 [ERR] Request validation failed
2019-04-17 12:35:23.136 -07:00 [INF] {"ClientId":null,"ClientName":null,"RedirectUri":null,"AllowedRedirectUris":null,"SubjectId":"anonymous","ResponseType":null,"ResponseMode":null,"GrantType":null,"RequestedScopes":"","State":null,"UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"clientwebapp","redirect_uri":"https://localhost:6001/signin-oidc","response_type":"code id_token","scope":"openid profile resource.api1 offline_access","response_mode":"form_post","nonce":"636911265221205762.NjRkNzczZGUtNzZiYS00MDAzLThiZDctMGEzYzAzMTczZDIxM2ZiZWQ1ZTAtNDM3MC00MjY5LWI3NTAtMjFlZGZiOWM2NmJj","state":"CfDJ8EEgl5mjJYhFtS8DfJca5DMMwYWMNQBxhUpdgEv3-jVDQ0lRjYpE1XnjX6Qfw_biWcbI4wp4xUi4n67096kmnRwzqRtgwgktn5Y5OYCg9RY22_lEAju5XG6uULAMBhSSE3dlvT-ZObxL8lVpFk1wi-WLUXbRAO5b9gEkjkiJObbgpzBRlY5EAPSfTjMOMzhYvITiQ-xrW9EYMual5-eRG9Br8fxqu4RIzK0HeiMUnTAboihUkB_eEa9p9oDQ-Auo9eCgMUsU7LDd5UKM8GBgIxfpKzHnyMND16q8ZgeV1fCvLCyNcj19wrsCLSgKzjHZ2u1anabVmX9BjoDQWVsV7kE"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:35:23.151 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":null,"RedirectUri":null,"Endpoint":"Authorize","SubjectId":null,"Scopes":"","GrantType":null,"Error":"unauthorized_client","ErrorDescription":"Unknown client or client not enabled","Category":"Token","Name":"Token Issued Failure","EventType":"Failure","Id":2001,"Message":null,"ActivityId":"0HLM3DEJD2MJ4:00000006","TimeStamp":"2019-04-17T19:35:23.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"TokenIssuedFailureEvent"}
2019-04-17 12:37:42.168 -07:00 [DBG] CORS request made for path: /connect/authorize from origin: https://localhost:6001 but was ignored because path was not for an allowed IdentityServer CORS endpoint
2019-04-17 12:37:42.182 -07:00 [DBG] Request path /connect/authorize matched to endpoint type Authorize
2019-04-17 12:37:42.184 -07:00 [DBG] Endpoint enabled: Authorize, successfully created handler: IdentityServer4.Endpoints.AuthorizeEndpoint
2019-04-17 12:37:42.191 -07:00 [INF] Invoking IdentityServer endpoint: IdentityServer4.Endpoints.AuthorizeEndpoint for /connect/authorize
2019-04-17 12:37:42.192 -07:00 [DBG] Start authorize request
2019-04-17 12:37:42.193 -07:00 [DBG] No user present in authorize request
2019-04-17 12:37:42.194 -07:00 [DBG] Start authorize request protocol validation
2019-04-17 12:37:42.678 -07:00 [DBG] clientwebapp found in database: true
2019-04-17 12:37:42.683 -07:00 [DBG] client configuration validation for client clientwebapp succeeded.
2019-04-17 12:37:42.689 -07:00 [DBG] Checking for PKCE parameters
2019-04-17 12:37:42.692 -07:00 [DBG] No PKCE used.
2019-04-17 12:37:42.932 -07:00 [DBG] Found ["openid","profile"] identity scopes in database
2019-04-17 12:37:43.021 -07:00 [DBG] Found [] API scopes in database
2019-04-17 12:37:43.024 -07:00 [ERR] Invalid scope: resource.api1
2019-04-17 12:37:43.025 -07:00 [ERR] Request validation failed
2019-04-17 12:37:43.025 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":"Client Web App","RedirectUri":"https://localhost:6001/signin-oidc","AllowedRedirectUris":["https://localhost:6001/signin-oidc"],"SubjectId":"anonymous","ResponseType":"code id_token","ResponseMode":"form_post","GrantType":"hybrid","RequestedScopes":"openid profile resource.api1 offline_access","State":"CfDJ8EEgl5mjJYhFtS8DfJca5DM73YfTYa_SAwLtx2flogwNWBRevouQDWimTSmVRk58sbcFj6Yu5ThJs2zr0Bk_R5vXLpk3mT5gpOovSGBjT89JjA9VSG_2fIGnyeo3Den48DzYqBPJxereBS8dwvppLDeBDg-_jJwAoLhTIb2eBrjgNVJSqWMNalimj5A4YOIio62UPfgtMULgNZrupPuxtxC9a7ZlGsiyS8S2iicD1slzyOYSljiiy1jZF9RXAVwMv9T_oef8OqW2otBOpI0BkZcadP8IOTA067eVI8X0mJNV3jFGXJdQGftH742lzh8fViKvhOe_e5mHOFna_ZUUXR0","UiLocales":null,"Nonce":null,"AuthenticationContextReferenceClasses":null,"DisplayMode":null,"PromptMode":null,"MaxAge":null,"LoginHint":null,"SessionId":null,"Raw":{"client_id":"clientwebapp","redirect_uri":"https://localhost:6001/signin-oidc","response_type":"code id_token","scope":"openid profile resource.api1 offline_access","response_mode":"form_post","nonce":"636911266616624384.NzcwNzY3NDUtZmJiNy00ZTk2LThiZjUtNzJjOGZhYzE3OGFhNmQ0M2VhM2UtNWRlNC00NTFjLWFhNDQtMTIzMjZhMWJmODY4","state":"CfDJ8EEgl5mjJYhFtS8DfJca5DM73YfTYa_SAwLtx2flogwNWBRevouQDWimTSmVRk58sbcFj6Yu5ThJs2zr0Bk_R5vXLpk3mT5gpOovSGBjT89JjA9VSG_2fIGnyeo3Den48DzYqBPJxereBS8dwvppLDeBDg-_jJwAoLhTIb2eBrjgNVJSqWMNalimj5A4YOIio62UPfgtMULgNZrupPuxtxC9a7ZlGsiyS8S2iicD1slzyOYSljiiy1jZF9RXAVwMv9T_oef8OqW2otBOpI0BkZcadP8IOTA067eVI8X0mJNV3jFGXJdQGftH742lzh8fViKvhOe_e5mHOFna_ZUUXR0"},"$type":"AuthorizeRequestValidationLog"}
2019-04-17 12:37:43.033 -07:00 [INF] {"ClientId":"clientwebapp","ClientName":"Client Web App","RedirectUri":"https://localhost:6001/signin-oidc","Endpoint":"Authorize","SubjectId":null,"Scopes":"openid profile resource.api1 offline_access","GrantType":"hybrid","Error":"invalid_scope","ErrorDescription":"Invalid scope","Category":"Token","Name":"Token Issued Failure","EventType":"Failure","Id":2001,"Message":null,"ActivityId":"0HLM3DEJD2MJ4:00000009","TimeStamp":"2019-04-17T19:37:43.0000000Z","ProcessId":3812,"LocalIpAddress":"::1:5000","RemoteIpAddress":"::1","$type":"TokenIssuedFailureEvent"}
2019-04-17 14:18:47.632 -07:00 [INF] Using the default authentication scheme Identity.Application for IdentityServer
2019-04-17 14:18:47.721 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for authentication
2019-04-17 14:18:47.727 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-in
2019-04-17 14:18:47.727 -07:00 [DBG] Using Identity.External as default ASP.NET Core scheme for sign-out
2019-04-17 14:18:47.737 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for challenge
2019-04-17 14:18:47.744 -07:00 [DBG] Using Identity.Application as default ASP.NET Core scheme for forbid
2019-04-17 14:18:49.441 -07:00 [FTL] Unable to start Kestrel.
System.IO.IOException: Failed to bind to address https://127.0.0.1:5000: address already in use. ---> Microsoft.AspNetCore.Connections.AddressInUseException: Only one usage of each socket address (protocol/network address/port) is normally permitted ---> System.Net.Sockets.SocketException: Only one usage of each socket address (protocol/network address/port) is normally permitted
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransport.BindAsync()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransport.BindAsync()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass22_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IServerAddressesFeature addresses, KestrelServerOptions serverOptions, ILogger logger, Func`2 createBinding)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
2019-04-17 14:18:49.544 -07:00 [FTL] An error occurred stopping the application
System.AggregateException: One or more errors occurred. (Not started. Call Start first.) ---> System.InvalidOperationException: Not started. Call Start first.
   at IdentityServer4.EntityFramework.TokenCleanup.Stop() in C:\local\identity\server4\EntityFramework.Storage\src\TokenCleanup\TokenCleanup.cs:line 77
   at Microsoft.Extensions.DependencyInjection.TokenCleanupHost.StopAsync(CancellationToken cancellationToken) in C:\ballen\github\identity\IdSvr4\IdentityServer4.EntityFramework\src\TokenCleanupHost.cs:line 37
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.<>c__DisplayClass4_0.<StopAsync>b__0(IHostedService service)
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.ExecuteAsync(Func`2 callback)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.ExecuteAsync(Func`2 callback)
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.StopAsync(CancellationToken token)
---> (Inner Exception #0) System.InvalidOperationException: Not started. Call Start first.
   at IdentityServer4.EntityFramework.TokenCleanup.Stop() in C:\local\identity\server4\EntityFramework.Storage\src\TokenCleanup\TokenCleanup.cs:line 77
   at Microsoft.Extensions.DependencyInjection.TokenCleanupHost.StopAsync(CancellationToken cancellationToken) in C:\ballen\github\identity\IdSvr4\IdentityServer4.EntityFramework\src\TokenCleanupHost.cs:line 37
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.<>c__DisplayClass4_0.<StopAsync>b__0(IHostedService service)
   at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.ExecuteAsync(Func`2 callback)<---

